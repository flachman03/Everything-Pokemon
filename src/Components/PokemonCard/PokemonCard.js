import React from 'react'
import './PokemonCard.scss'

export const PokemonCard = ({ pokemon }) => {
  return (
    <article className="card">
      <img src={pokemon.sprites.front_default} alt={pokemon.name} className="card__image"/>
      <h2 className="card__title">{pokemon.name}</h2>

    </article>
  )
}