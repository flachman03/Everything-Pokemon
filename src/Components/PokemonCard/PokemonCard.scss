.card {
  width: 150px;
  height: 145px;
  margin: 10px;
  border-radius: 5px;
  background: lightblue;
  text-decoration: none;
  &:hover {
    background: hsla(254, 0%, 0%, 0.7);

  }
}

.card__image {
  height: 120px;
  width: 120px;
}

.card__title {
  position: relative;
  z-index: 1;
  bottom: 30px; 
  color: white;
}

.styling {
  background: lightblue;
}