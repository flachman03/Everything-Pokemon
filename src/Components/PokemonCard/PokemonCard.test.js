import React from 'react'
import { shallow } from 'enzyme'
import { PokemonCard } from './PokemonCard'

describe('PokemonCard', () => {
  let wrapper, mockData;
  beforeEach(() => {
    mockData = {name: 'charmander', sprites: {}}
    wrapper = shallow(<PokemonCard pokemon={mockData} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have an article named card', () => {
    expect(wrapper.find('.card')).toHaveLength(1)
  })

  it('should have img named card__image', () => {
    expect(wrapper.find('.card__image')).toHaveLength(1)
  })

  it('should have h2 named card__title', () => {
    expect(wrapper.find('.card__title')).toHaveLength(1)
  })
})