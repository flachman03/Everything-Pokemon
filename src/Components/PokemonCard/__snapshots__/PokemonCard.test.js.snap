// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard should match the snapshot 1`] = `ShallowWrapper {}`;
